{{cookiecutter.project_slug}}/sonar-project.properties

```
@@ -0,0 +1,18 @@
+{%- set username = "None" if cookiecutter.remote_url == "None" else cookiecutter.remote_url.split("/")[-2].split(":")[-1] -%}
+{%- set remote_slug = "None" if cookiecutter.remote_url == "None" else cookiecutter.remote_url.replace(".git", "").split("/")[-1] -%}
+sonar.projectKey={{ username }}_{{ remote_slug }}
+sonar.organization={{ username }}
+sonar.host.url=https://sonarcloud.io
+
+# Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
+sonar.sources=./src
+
+# Exclusions
+sonar.coverage.exclusions=tests/*,ext/*
+sonar.test.exclusions=tests/*
+
+# Properties specific to the C/C++ analyzer:
+sonar.cfamily.threads=2
+sonar.cfamily.cache.enabled=false
```

enabling this option is probably worthwhile, then it caches the previous analysis and only re-analyzes changed source files with each commit (otherwise the sonar analysis can end up being a significant bottleneck in the CI as the codebase grows)
```suggestion
sonar.cfamily.cache.enabled=true
```
on the github actions side it would just require adding these lines (or similar) to the yml:
https://github.com/spatial-model-editor/spatial-model-editor/blob/master/.github/workflows/quality.yml#L39-L44